### PR TITLE
doc: Add placeholder for migration notes

### PR DESCRIPTION
**Pull request by umapraseeda**
Add a working draft for the migration notes for 2.5

### CODE REVIEW

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.4_to_2.5.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
+:orphan:
+
+.. _migration_2.5:
+
+Migration guide for |NCS| v2.5.0 (Working draft)
+################################################
+
+This document describes the changes required or recommended when migrating your application from |NCS| v2.4.0 to |NCS| v2.5.0.
```
Comment (wbober) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.4_to_2.5.rst:8]: Perhaps we should explain what is the intent behind required and recommended changes? For required it seems to be pretty clear: one needs these changes to make the code build or perform in the same way as in previous release.

But what is the reasoning for the recommended changes?

Comment (umapraseeda) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.4_to_2.5.rst]: Added this.

File: doc/nrf/releases_and_maturity.rst [hunk 2]
```diff
@@ -27,3 +27,15 @@ If an issue is found in a release after it has taken place, those issues are lis
    releases_and_maturity/repository_revisions
    releases_and_maturity/software_maturity
    releases_and_maturity/known_issues
+
+Migration guides
+****************
+
+The |NCS| provides migration guides for all minor releases, in order to assist user's transition from the previous release.
```
Comment (balaji-nordic) [inline doc/nrf/releases_and_maturity.rst:34]: We intend to provide the guides for major releases too right?

**Review by balaji-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity.rst (hunk 2 above)
Comment (umapraseeda) [reply doc/nrf/releases_and_maturity.rst]: Modified

**Review by balaji-nordic — APPROVED** [review]

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst [hunk 3]
```diff
@@ -0,0 +1,32 @@
+:orphan:
+
+.. _migration_2.5:
+
+Migration guide for |NCS| v2.5.0 (Working draft)
+################################################
+
+This document describes the changes required or recommended when migrating your application to |NCS| v2.5.0.
```
Comment (shanthanordic) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst:8]: to rephrase "to NCS 2.5.0 from NCS 2.4.0"

Comment (umapraseeda) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst:8]: This was originally phrased like this in [one of the previous commits](https://github.com/nrfconnect/sdk-nrf/commit/bc5fd287f9512f6aca5dd7274bd86061a285665e#diff-9a54356848c52708ddb8709157b048cf117565530c3b2a9ce982905e69b20768R8), rephrased to its current form after @carlescufi 's suggestion.

Comment (carlescufi) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst:8]: I agree with @shanthanordic. I suggest we write:
```suggestion
This document describes the changes required or recommended when migrating your application from |NCS| v2.4.0 to |NCS| v2.5.0.
```

**Review by carlescufi — CHANGES_REQUESTED** [review]
see my latest comment

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst (hunk 3 above)
Comment (umapraseeda) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst]: Updated

**Comment by umapraseeda** [comment]
> see my latest comment

Fixed.

**Review by shanthanordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**